ledger-core/virtual/execute/execute.go

```diff
@@ -722,6 +722,8 @@ func (s *SMExecute) stepSendOutgoing(ctx smachine.ExecutionContext) smachine.Sta
 		}
 	} else {
 		if s.outgoingSentCounter >= MaxOutgoingSendCount {
+			// TODO when CallSummary will live longer than one pulse it needs to be updated
+			s.sendDelegatedRequestFinished(ctx, nil)
```

Comment:
I don't like the idea of setting "sendDelegatedRequestFinished" in every place where work can be stopped prematurely :)
I'd prefer to set "ErrorHandler"/"some kind of destructors" somewhere and catch all that situations at once.

Reply:
We discuss this with @ruz and decided that this is important for pending tables. Or I can't understand your idea in second sentence.

Reply:
I mean that part of code (send pending request finished) must be executed on every exit scenario after we've received VDelegatedCallRequest

Reply:
But that's a big addition, so I guess only if @ruz is OK

Reply:
Yeah, we need some kind of destructor for this.